tests/providers/amazon/aws/deferrable/triggers/test_batch.py

```diff
@@ -0,0 +1,131 @@
+        region_name="eu-west-1",
+        aws_conn_id="airflow_test",
+    )
+
+    def test_batch_trigger_serialization(self):
+        """
+        Asserts that the BatchOperatorTrigger correctly serializes its arguments
+        and classpath.
+        """
+
+        classpath, kwargs = self.TRIGGER.serialize()
+        assert classpath == "airflow.providers.amazon.aws.triggers.batch.BatchOperatorTrigger"
+        assert kwargs == {
+            "job_id": JOB_ID,
+            "job_name": JOB_NAME,
+            "job_definition": "hello-world",
+            "job_queue": "queue",
+            "waiters": None,
+            "tags": {},
+            "max_retries": MAX_RETRIES,
+            "status_retries": STATUS_RETRIES,
+            "parameters": {},
+            "overrides": {},
+            "array_properties": {},
+            "region_name": "eu-west-1",
+            "aws_conn_id": "airflow_test",
+        }
+
+    @pytest.mark.asyncio
+    async def test_batch_trigger_run(self):
+        """Test that the task is not done when event is not returned from trigger."""
+
+        task = asyncio.create_task(self.TRIGGER.run().__anext__())
+        await asyncio.sleep(0.5)
+        # TriggerEvent was not returned
+        assert task.done() is False
+
+    @pytest.mark.asyncio
+    @async_mock.patch("airflow.providers.amazon.aws.hooks.batch_client.BatchClientAsyncHook.monitor_job")
+    async def test_batch_trigger_completed(self, mock_response):
+        """Test if the success event is  returned from trigger."""
+        mock_response.return_value = {"status": "success", "message": f"AWS Batch job ({JOB_ID}) succeeded"}
+
+        generator = self.TRIGGER.run()
+        actual_response = await generator.asend(None)
+        assert (
+            TriggerEvent({"status": "success", "message": f"AWS Batch job ({JOB_ID}) succeeded"})
+            == actual_response
+        )
+
+    @pytest.mark.asyncio
+    @async_mock.patch("airflow.providers.amazon.aws.hooks.batch_client.BatchClientAsyncHook.monitor_job")
+    async def test_batch_trigger_failure(self, mock_response):
+        """Test if the failure event is returned from trigger."""
+        mock_response.return_value = {"status": "error", "message": f"{JOB_ID} failed"}
+
+        generator = self.TRIGGER.run()
+        actual_response = await generator.asend(None)
+        assert TriggerEvent({"status": "error", "message": f"{JOB_ID} failed"}) == actual_response
+
+    @pytest.mark.asyncio
+    @async_mock.patch("airflow.providers.amazon.aws.hooks.batch_client.BatchClientAsyncHook.monitor_job")
+    async def test_batch_trigger_none(self, mock_response):
+        """Test if the failure event is returned when there is no response from hook."""
+        mock_response.return_value = None
+
+        generator = self.TRIGGER.run()
+        actual_response = await generator.asend(None)
+        assert TriggerEvent({"status": "error", "message": f"{JOB_ID} failed"}) == actual_response
```

Comment:
Similar comment. By using `@pytest.mark.parametrize`, you can merge these 3 tests into one

Reply:
Even I had parameterised the test at first but got this error https://github.com/Martiusweb/asynctest/issues/152#issue-666326338 and CI was failing, by looking at this PR comment https://github.com/apache/airflow/pull/29801#discussion_r1125620257 I had them separated then the CI was passing for me.